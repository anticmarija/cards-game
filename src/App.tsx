import React from 'react';
import './App.css';
import GameIntro from './GameIntro/GameIntro';
import { connect } from 'react-redux';
import { getNumberOfPlayers } from './store/reducers/game.reducer';

interface Props {
  numberOfPlayers?: number;
}

const App: React.FC<Props> = ({ numberOfPlayers }) => {
  console.log(numberOfPlayers);
  return (
    <div className='game'>{numberOfPlayers ? 'llallala' : <GameIntro />}</div>
  );
};

export const mapStateToProps = (state: any) => ({
  numberOfPlayers: getNumberOfPlayers(state),
});

export default connect(mapStateToProps)(App);
