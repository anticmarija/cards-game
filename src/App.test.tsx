import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('Test Layout component', () => {
  it('should render correctly', () => {
    const { container } = render(<App />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
