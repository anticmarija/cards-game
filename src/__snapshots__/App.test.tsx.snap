// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout component should render correctly 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      class="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
